### PR TITLE
[LLVM][IR] Teach constant integer binop folds about vector ConstantInts.

### PR DESCRIPTION
**Pull request by paulwalker-arm**
The existing logic mostly works with the main changes being:
 * Use getScalarSizeInBits instead of IntegerType::getBitWidth
 * Use ConstantInt::get(Type* instead of ConstantInt::get(LLVMContext

### CODE REVIEW

File: llvm/test/Transforms/InstCombine/or.ll [hunk 1]
```diff
@@ -399,10 +400,15 @@ define i32 @test30(i32 %A) {
...
-; CHECK-NEXT:    ret <2 x i32> [[E]]
+; CONSTVEC-LABEL: @test30vec(
+; CONSTVEC-NEXT:    [[TMP1:%.*]] = and <2 x i32> [[A:%.*]], splat (i32 -58312)
+; CONSTVEC-NEXT:    [[E:%.*]] = or disjoint <2 x i32> [[TMP1]], splat (i32 32962)
+; CONSTVEC-NEXT:    ret <2 x i32> [[E]]
+;
+; CONSTSPLAT-LABEL: @test30vec(
+; CONSTSPLAT-NEXT:    [[D:%.*]] = and <2 x i32> [[A:%.*]], splat (i32 -58312)
+; CONSTSPLAT-NEXT:    [[E:%.*]] = or disjoint <2 x i32> [[D]], splat (i32 32962)
+; CONSTSPLAT-NEXT:    ret <2 x i32> [[E]]
```
Comment (nikic) [inline llvm/test/Transforms/InstCombine/or.ll:411]: Is the difference here only in variable naming?

Comment (paulwalker-arm) [reply llvm/test/Transforms/InstCombine/or.ll:411]: Yes.  My guess is that we're now going down a path where the original name is preserved.  I was going to ignore it but then remembered how it bugs me when I run the update scripts and end up with a bunch of changes unrelated to my PR.

Comment (nikic) [reply llvm/test/Transforms/InstCombine/or.ll:411]: Nowadays the update scripts are supposed to keep the check names stable even if the generated names change :)

**Review by nikic — APPROVED** [review]
LGTM

**Pull request merged**